### PR TITLE
[IOS-347]: Cache Browse page objects

### PR DESCRIPTION
**Pull request by yuliaveres**
The Browse objects are stored in cache now, and updated each time on fetch. Cached data is display prior to fetched.

### CODE REVIEW

File: Toshi/APIClients/AppsAPIClient.swift [hunk 1]
```diff
@@ -32,55 +40,78 @@ public class AppsAPIClient: NSObject, CacheExpiryDefault {
...
+    private lazy var cache: Cache<TokenUsersCacheData> = {
+        do {
+            return try Cache<TokenUsersCacheData>(name: "appsCache")
+        } catch {
+            fatalError("Couldn't instantiate the apps cache")
+        }
+    }()
+
+    func getTopRatedApps(limit: Int = 10, cachedCompletion: @escaping TokenUserResults, fetchedCompletion: @escaping TokenUserResults) {
+        if let data = self.cache.object(forKey: topRatedAppsCachedDataKey) as TokenUsersCacheData?, let ratedUsers = data.objects as [TokenUser]? {
```
Comment (roberthein) [inline Toshi/APIClients/AppsAPIClient.swift]: too much explicit self's in this file

File: Toshi/APIClients/idAPIClient.swift [hunk 2]
```diff
@@ -385,60 +399,74 @@ import Teapot
         }
     }
 
-    public func getTopRatedPublicUsers(limit: Int = 10, completion: @escaping (_ apps: [TokenUser], _ error: Error?) -> Void) {
+    public func getTopRatedPublicUsers(limit: Int = 10, cachedCompletion: @escaping TokenUserResults, fetchedCompletion: @escaping TokenUserResults) {
 
-        self.teapot.get("/v1/search/user?public=true&top=true&recent=false&limit=\(limit)") { (result: NetworkResult) in
+        if let data = self.cache.object(forKey: topRatedUsersCachedDataKey) as TokenUsersCacheData?, let ratedUsers = data.objects as [TokenUser]? {
```
Comment (roberthein) [inline Toshi/APIClients/idAPIClient.swift]: too much explicit self's in this file

File: Toshi/APIClients/AppsAPIClient.swift [hunk 3]
```diff
@@ -32,55 +40,78 @@ public class AppsAPIClient: NSObject, CacheExpiryDefault {
...
-        self.teapot.get("/v1/search/apps?top=true&recent=false&limit=\(limit)") { (result: NetworkResult) in
+    private lazy var cache: Cache<TokenUsersCacheData> = {
+        do {
+            return try Cache<TokenUsersCacheData>(name: "appsCache")
+        } catch {
+            fatalError("Couldn't instantiate the apps cache")
+        }
+    }()
+
+    func getTopRatedApps(limit: Int = 10, cachedCompletion: @escaping TokenUserResults, fetchedCompletion: @escaping TokenUserResults) {
```
Comment (roberthein) [inline Toshi/APIClients/AppsAPIClient.swift]: I don't understand why there are 2 completion blocks that do exactly the same. 
I think you can replace
```
cachedCompletion: @escaping TokenUserResults, fetchedCompletion: @escaping TokenUserResults
```
with:
```
completion: @escaping TokenUserResults
```
everywhere.

Comment (yuliaveres) [reply Toshi/APIClients/AppsAPIClient.swift]: i need different completions for different results - cached first - then fetched. cached are always instant, after fetch is complete (or if it is complete at all) - `fetchedCompletion` is called and data is refreshed in a cell .

Comment (roberthein) [reply Toshi/APIClients/AppsAPIClient.swift]: But you can just call the same `completion` block twice.

Comment (roberthein) [reply Toshi/APIClients/AppsAPIClient.swift]: there is no difference between the blocks right?

Comment (yuliaveres) [reply Toshi/APIClients/AppsAPIClient.swift]: 🤔  , no there is no, but i am wondering if i can. 
ok i will change to  one `completion`

Comment (roberthein) [reply Toshi/APIClients/AppsAPIClient.swift]: 👌

**Pull request merged**